Parser error: Misleading error messages when function declaration has '->' without return type

## Description

When a function is declared with the return arrow `->` but without a return type, the parser throws misleading error messages instead of identifying the actual syntax error.

## Example Code

```ez
do main() -> {
    using std
    // rest of function
}
```

## Current Behavior

The parser throws two errors:

```
error[E1002]: expected {, got USING instead
  --> examples/tests.ez:90:5
   |
90 |     using std
   |     ^^^^^ expected token not found
   |

error[E1003]: expected expression, found end of block
  --> examples/tests.ez:110:1
   |
110 | }
   | ^ invalid syntax
```

## Expected Behavior

The parser should recognize that the `->` return arrow is present but no return type has been specified, and provide a clear error message such as:

```
error[EXXXX]: expected return type after '->'
  --> examples/tests.ez:XX:XX
   |
XX | do main() -> {
   |           ^^ return type missing
   |
```

## Notes

The current errors are technically triggered correctly (the parser gets confused after the malformed declaration), but they don't help the user understand what they did wrong. The root cause is the missing return type after `->`.

## Impact

This causes confusion for users who will try to fix the wrong part of their code based on the misleading error messages.

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -1056,15 +1056,28 @@ func (p *Parser) parseFunctionDeclarationWithAttrs(attrs []*Attribute) *Function
 
 	// Parse return type(s)
 	if p.peekTokenMatches(ARROW) {
+		arrowToken := p.peekToken // save for error reporting
 		p.nextToken() // consume ->
 		p.nextToken() // move to return type
+
+		// Check if return type is missing (e.g., "do func() -> {")
+		if p.currentTokenMatches(LBRACE) {
+			msg := "expected return type after '->'"
+			p.addEZError(errors.E1002, msg, arrowToken)
+			return nil
+		}
 
 		if p.currentTokenMatches(LPAREN) {
 			// Multiple return types
 			stmt.ReturnTypes = p.parseReturnTypes()
-		} else {
+		} else if p.currentTokenMatches(IDENT) {
 			// Single return type
 			stmt.ReturnTypes = []string{p.currentToken.Literal}
+		} else {
+			// Invalid token after arrow
+			msg := fmt.Sprintf("expected return type after '->', got %s instead", p.currentToken.Type)
+			p.addEZError(errors.E1002, msg, arrowToken)
+			return nil
 		}
 	}
 

